aspnetcore/security/authentication/social/additional-claims/samples/2.x/ClaimsSample/Startup.cs

```diff
@@ -30,27 +27,31 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.Configure<CookiePolicyOptions>(options =>
             {
+                // This lambda determines whether user consent for non-essential cookies is needed for a given request.
                 options.CheckConsentNeeded = context => true;
                 options.MinimumSameSitePolicy = SameSiteMode.None;
             });
 
-            services.AddDbContext<AppDbContext>(options => 
+            services.AddDbContext<ApplicationDbContext>(options => 
                 options.UseInMemoryDatabase("InMemoryDb"));
 
-            services.AddIdentity<ApplicationUser, IdentityRole>()
-                .AddEntityFrameworkStores<AppDbContext>()
-                .AddDefaultTokenProviders();
-
             #region snippet_AddGoogle
             services.AddAuthentication().AddGoogle(options =>
             {
                 // Provide the Google Client ID
-                options.ClientId = "XXXXXXXXXXX.apps.googleusercontent.com";
-                // Provide the Google Secret
-                options.ClientSecret = "g4GZ2#...GD5Gg1x";
-                options.Scope.Add("https://www.googleapis.com/auth/plus.login");
-                options.ClaimActions.MapJsonKey(ClaimTypes.Gender, "gender");
+                options.ClientId = "XXXXXXXXXXXXXXX.apps.googleusercontent.com";
+                // Register with User Secrets using:
+                // dotnet user-secrets set "Authentication:Google:ClientId" "{Client ID}"
+
+                // Provide the Google Client Secret
+                options.ClientSecret = "{Client Secret}";
+                // Register with User Secrets using:
+                // dotnet user-secrets set "Authentication:Google:ClientSecret" "{Client Secret}"
+
+                options.ClaimActions.MapJsonKey("urn:google:picture", "picture", "url");
+                options.ClaimActions.MapJsonKey("urn:google:locale", "locale", "string");
                 options.SaveTokens = true;
+
                 options.Events.OnCreatingTicket = ctx =>
                 {
                     List<AuthenticationToken> tokens = ctx.Properties.GetTokens() 
```

Comment:
This example shouldn't be casting. Calling ToList() would be better.

Reply:
From the engineering sample ...

https://github.com/aspnet/AspNetCore/blob/658b37d2bd3b0da2c07e25b53fa5c97ce8d656d3/src/Security/samples/Identity.ExternalClaims/Startup.cs#L49

@HaoK?